jenkins/opensearch/integ-test.jenkinsfile

```diff
@@ -80,66 +96,78 @@ pipeline {
+                    def componentList = COMPONENT_NAME ? COMPONENT_NAME.trim().split(" ") as List : componentDefaultList
+                    String switch_user_non_root = (distribution.equals('rpm') || distribution.equals('deb')) ? 'true' : 'false'
+                    echo "switch_user_non_root: ${switch_user_non_root}"
                     echo "componentList: ${componentList}"
 
+                    for (component_check in componentList) {
+                        if (! componentDefaultList.contains(component_check)) {
+                           error("${component_check} is not in build manifest: ${componentDefaultList}, exit 1") 
+                        }
+                    }
+
+                    echo "Downloading from S3: ${artifactPath}"
+                    downloadFromS3(
+                        assumedRoleName: 'opensearch-bundle',
+                        roleAccountNumberCred: 'jenkins-aws-account-public',
+                        downloadPath: "${artifactPath}/",
+                        bucketName: "${ARTIFACT_BUCKET_NAME}",
+                        localPath: "${WORKSPACE}/artifacts",
+                        force: true
+                      )
+                    sh("mv -v $WORKSPACE/artifacts/${artifactPath} $WORKSPACE")
+
+                    // Stash the current working directory files, aka opensearch-build repo
+                    // Unstash later in each triggered stage to run integTest
+                    stash includes: "**", name: "opensearch-build-repo-$BUILD_NUMBER"
+
                     componentTests = [:]
 
                     for (component in componentList) {
                         // Must use local variable due to groovy for loop and closure scope
                         // Or the stage will be fixed to the last item in return when new stages are triggered here
                         // https://web.archive.org/web/20181121065904/http://blog.freeside.co/2013/03/29/groovy-gotcha-for-loops-and-closure-scope/
-                        def local_component = component
-                        def wait_seconds = componentList.indexOf(local_component) * 20
+                        def local_component = component.trim()
+                        def local_component_index = componentList.indexOf(local_component)
+                        def wait_seconds = local_component_index * 10
 
                         echo "Add Component: ${local_component}"
                         componentTests["Run Integtest ${local_component}"] = {
                             // Using scripted pipelines to trigger dynamic parallel stages
                             timeout(time: 2, unit: 'HOURS') {
                                 node(AGENT_LABEL) {
-                                    docker.image(DOCKER_AGENT.image).inside(DOCKER_AGENT.args) {
+                                    docker.image(docker_images["$distribution"]).inside(docker_args["$distribution"]) {
```

Comment:
Didn't recall and just want to make sure as `distribution` is specified as environmental variable, does this also work? or it was `env.distribution`?

Reply:
You can just call distribution it will take the env var.